### PR TITLE
Improve DockerFile script engine

### PR DESCRIPTION
**Pull request by fviale**
- Allow defining the image and container name as generic information
 - Allow defining the actions to execute as generic information (default is build,run,stop,rmi
 - Introduce the exec action, which allows to do docker exec inside a running container
 - Add relevant unit tests

### CODE REVIEW

**Comment by fviale** [comment]
Retest this please

File: src/main/java/jsr223/docker/file/DockerFileScriptEngine.java [hunk 1]
```diff
@@ -394,11 +497,15 @@ private void handleShutdown(ScriptContext context) {
...
+        if (processExec != null) {
+            processExec.destroy();
+        }
+
+        if (containerStarted && dockerActions.contains("stop")) {
             stopAndRemoveContainer(containerTagName, context);
         }
 
-        if (imageCreated && !DockerFilePropertyLoader.getInstance().isKeepDockerFile()) {
+        if (imageCreated && dockerActions.contains("rmi")) {
```
Comment (MaelAudren) [inline src/main/java/jsr223/docker/file/DockerFileScriptEngine.java]: That would be nice to create final variables for "rmi" and "stop".

File: src/main/java/jsr223/docker/file/DockerFileScriptEngine.java [hunk 2]
```diff
@@ -196,44 +216,27 @@ public void run() {
...
-                                                           context.getErrorWriter(),
-                                                           context.getReader());
-
-            // Wait for process build to exit
-            int exitValueBuild = processBuild.waitFor();
+            int exitValue = 0;
 
-            if (exitValueBuild != 0) {
-                throw new ScriptException("Docker File Build failed with exit code " + exitValueBuild);
+            if (dockerActions.contains("build")) {
```
Comment (MaelAudren) [inline src/main/java/jsr223/docker/file/DockerFileScriptEngine.java]: It would be nice to have a final variable for "build"

**Review by MaelAudren — APPROVED** [review]

File: src/main/java/jsr223/docker/file/DockerFileScriptEngine.java (hunk 1 above)
Comment (fviale) [reply src/main/java/jsr223/docker/file/DockerFileScriptEngine.java]: okay

**Review by aminelouati — APPROVED** [review]

**Pull request merged**